### PR TITLE
Add support for History operation

### PR DESCRIPTION
**Pull request by jhlee-mitre**
Major changes:

1. Added TestScripts and fixtures for Observation Resource
2. Added TestScript and testing codes for history
3. Updated read.me

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+## About the Project
+
+The TestScript Execution Engine is to support FHIR testing in an integrated and systematic way by providing the following features.
```
Comment (Jack-Fraser16) [inline README.md:3]:
```suggestion
The TestScript Engine is to support FHIR testing in an integrated and systematic way by providing the following features.
```
Let's just keep it at 'TestScript Engine' for now.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+
+* A general purpose test execution tool to be implemented in a variety of use cases
+* Support TestScript / TestReport Resource types as inputs and outputs to be aligned with FHIR architecture
+* (TBA) Support an end-to-end pipeline of FHIR testing including authoring exception handling and analytics
```
Comment (Jack-Fraser16) [inline README.md:7]: I'll defer to @jawalonoski on whether or not we want to advertise this intention

**Review by Jack-Fraser16 — CHANGES_REQUESTED** [review]
- I would rename this PR to be more descriptive about what it's accomplishing. Something like 'Add support for History operation'. 
- We should hide the TestReports directory from git. I think it's excessive to push up every TestReport we create -- especially if running the engine will just recreate them anyways. 
- The opposite for TestScripts -- let's keep pushing up every new, unique TestScript we find/create.
- We need to solidify on our naming conventions for both TestScripts and TestReports. I know that probably seems excessive and semantic at this point, but I think it's gonna help keep us organized in the long run. I'll start a Slack thread about it.  
- Last, we need to get back in the habit of writing unit tests. The ones we have are .... less than stellar. I'm going to get you started on some unit tests specifically for `history` and then push them up and let you take them over the finish line @jhlee-mitre. And then in the near future, you'll have the joy of rewriting our existing tests, which will help us cover our bases and confirm that what we have works well.

Thrilled to have you working on this and keep pushing it forward 👍

File: lib/TestScriptRunnable.rb [hunk 3]
```diff
@@ -116,18 +117,24 @@ def run client = nil
...
       request_type = REQUEST_TYPES[op.local_method || op.type.code]
       throw :exit, report.skip('notImplemented') unless request_type
 
       request = [request_type, extract_path(op), extract_body(request_type, op), extract_headers(op)]
       request.compact!
 
       begin
-        reply = client.send *request
+        if op.type.code == 'history'
+          reply = client.resource_instance_history(op.class, id_map[op.targetId])
```
Comment (Jack-Fraser16) [inline lib/TestScriptRunnable.rb:134]:
```suggestion
          reply = client.resource_instance_history(op.class, id_map[op.targetId])
```
I'm going to take a closer look at this while starting the unit tests. The way the request array is created, we shouldn't have to interrogate the operation for its type -- every type should just work. So, if that's not the case, we need to figure out why and how to fix the way 'request' is constructed.

Comment (jhlee-mitre) [reply lib/TestScriptRunnable.rb:134]: I wasn't sure how 'client and.send' and 'request' cover history request, couldn't find an example of code. I agree with you that one method should be abstracted enough to cover all types from a class.

File: README.md (hunk 2 above)
Comment (jhlee-mitre) [reply README.md:7]: Sure, I believe there will be rooms for polishing and refining the introduction as to where the project scope and direction go.

**Comment by jhlee-mitre** [comment]
Accepted Jack's suggestions. Read.me content has rooms for improvement.

**Pull request merged**